crates/oxc_linter/src/config/oxlintrc.rs

```diff
@@ -152,6 +159,23 @@ impl Oxlintrc {
 
         config.path = path.to_path_buf();
 
+        if let Some(config_dir) = config.path.parent() {
+            if let Some(external_plugins) = &mut config.external_plugins {
+                *external_plugins = std::mem::take(external_plugins)
+                    .into_iter()
+                    .map(|(_, specifier)| (Some(config_dir.to_path_buf()), specifier))
+                    .collect();
+            }
+
+            for override_config in config.overrides.iter_mut() {
+                if let Some(external_plugins) = &mut override_config.external_plugins {
+                    *external_plugins = std::mem::take(external_plugins)
+                        .into_iter()
+                        .map(|(_, specifier)| (Some(config_dir.to_path_buf()), specifier))
+                        .collect();
+                }
+            }
+        }
```

Comment:
Unless I'm misunderstanding, `config.path.parent().unwrap()` should be safe to use here. We just read the config file, which I assume means it must have a parent directory - if `path` was `/` (or Windows equivalent) we would have already exited because you can't read a directory as a file.

I *think* that means we can make `external_plugins` `Option<FxHashSet<(PathBuf, String)>>` (no `Option` around the `PathBuf`). We'd just need to create dummy `PathBuf`s in the deserializer (ergh serde is annoying).